### ISSUE TITLE
A11y: [Chats] Context menu (long-press) not accessible via exter...

### ISSUE THREAD

**Issue by mxandreas** [element-hq/element-x-android#6398]
- **Checkpoint ID:** 11.2.1.1/4.1
- **Component:** Chats

### Issue Description
Context menu (long-press) not accessible via external keyboard.

### Recommendation
Provide keyboard alternative for long-press actions.

--- 
*Report Date: 25.02.2026*

**Comment by bmarty** [comment]
This has been implemented in https://github.com/element-hq/element-x-android/pull/5120 "Shift + F10" should open the context menu when a room is focused. Maybe it does not work, maybe the feature is not that common, so can hardly be discovered.

**Comment by mxandreas** [comment]
Details from the report:

_A context menu can be opened with the screen reader using a double-tap and hold, or by long-pressing with a finger. However, there is no keyboard-operable way to open this context menu. It is recommended to provide a keyboard-accessible alternative for all actions currently available via long press, for example by oﬀering a focusable “More options” button or another pattern that exposes the same actions without requiring a long press._

So, I guess that is not that common, but I think it is fine to keep it as it is when Shift+F10 works.

**Comment by michaelnissenbaum** [comment]
This keyboard shortcut (Shift + F10) is at least not known to me. As I already wrote in the report, from my perspective it would be better to provide an alternative, for example a button.

**Comment by mxandreas** [comment]
@americanrefugee Can you please look into the alternative, e.g. if/where/how that button could exist.

**Comment by americanrefugee** [comment]
Is it possible to slide a room list item to the left/right via an external keyboard to expose additional buttons?

Like on WA, Telegram, and Signal...
- Swipe right > all expose Read/Unread and Pin buttons
- Swipe left > exposes other buttons like Delete, Mute, Archive, and an overflow button for more options

We can do something similar, and it would be a common pattern.

However, if we can't offer such controls with an external keyboard then we have the same problem. 

Meanwhile, displaying a tappable button in each room cell that shows the context menu would negatively impact the appearance/simplicity of room cells. I would not recommend this.

**Comment by mxandreas** [comment]
> Is it possible to slide a room list item to the left/right via an external keyboard to expose additional buttons?

I doubt that there is a standard keyboard action for the sliding/wiping or if there is, then how many people are familiar with that? Happy to be wrong here.

However, that is kind of the reason why we're looking for a "regular button" because that is something that everybody knows how to use with an external keyboard.

**Comment by americanrefugee** [comment]
As stated, a dedicated button to show the content menu for each and every room list item is going to take up a lot of room and visually clutter the room list. At least on iOS, all the messaging apps require you to swipe left/right to reveal more buttons.

**Comment by bmarty** [comment]
The long press actions displayed after a long click on a room are also available in the room detail screen, except for "Mark as unread" (and clear cache of the room, but this is only a developer thing).

So maybe adding the missing action "Mark as unread" to the room detail settings can ensure that the actions are available to everyone, even without easy access to the long press menu?

**Comment by americanrefugee** [comment]
@mxandreas and I think that's a good solution @bmarty

Here are the designs for Room Info...

---

1. If **NO unreads** = Show new menu item "Mark as unread"
   - [iOS](https://www.figma.com/design/jvZ7TIyd7V9Vn8sIDB9vMd/Compound-iOS-Templates?node-id=3857-254494&t=cOM6F92lDtae2fZ8-4)
   - [Android](https://www.figma.com/design/pDlJZGBsri47FNTXMnEdXB/Compound-Android-Templates?node-id=21-120235&t=InLC3zDmsIdEqMbj-4)

<img width="360" height="800" alt="Image" src="https://github.com/user-attachments/assets/abc42c24-9c94-4fd6-9850-60fd3bbcb6c7" />

---

2. If **YES unreads** = Show net menu item "Mark as read" with an indicator
   - [iOS](https://www.figma.com/design/jvZ7TIyd7V9Vn8sIDB9vMd/Compound-iOS-Templates?node-id=3857-255344&t=cOM6F92lDtae2fZ8-4)
   - [Android](https://www.figma.com/design/pDlJZGBsri47FNTXMnEdXB/Compound-Android-Templates?node-id=21-120243&t=InLC3zDmsIdEqMbj-4)

<img width="360" height="800" alt="Image" src="https://github.com/user-attachments/assets/6c752940-6fa6-4bd3-9fc0-628916277b8b" />
